pkg/health/health.go

```diff
@@ -78,21 +88,33 @@ func StartEndpointHealthCheck(ctx context.Context, endpointAddress string, opts
 			}
 		}
 
+		// Initial state is unhealthy until we validate it
+		// This makes it so the channel will receive a healthy status on the first successful healthcheck
+		failureCount := &atomic.Int32{}
+		failureCount.Store(options.failureThreshold)
+		signalChan <- false
+
 		for {
 			select {
 			case <-ch:
-				go doHealthCheck(client, endpointAddress, options.requestTimeout, signalChan, failureCount, options)
+				wg.Add(1)
+				go func() {
+					doHealthCheck(client, options.address, options.requestTimeout, signalChan, failureCount, options)
+					wg.Done()
+				}()
```

Comment:
Why is this done in a go routine? It doesn't look like it should be as it should only happen one at a time. We are also blocking from this with the wait group for a context cancelled.

Reply:
I tried removing the goroutine but stuff (including tests) broke. I will leave it there for now, since it was there before too (without the waitgroup but that doesn't matter).